src/compiler/sys.ts

```diff
@@ -100,6 +109,34 @@ module ts {
                 }
             }
 
+            function getNames(collection: any): string[] {
+                var result: string[] = [];
+                for (var e = new Enumerator(collection); !e.atEnd(); e.moveNext()) {
+                    result.push(e.item().Name);
+                }
+                return result.sort();
+            }
+
+            function readDirectory(path: string, extension?: string): string[] {
+                var result: string[] = [];
+                visitDirectory(path);
+                return result;
+                function visitDirectory(path: string) {

```

Comment:
I thought we don't use functions within functions anymore because of extra allocations